False positive Non-iterable value with async comprehensions

### Steps to reproduce
Copy this code in `test.py`
```python
"""Some module docstring"""

async def get_somethings():
    """Some docstring"""
    for something in range(10):
        yield something

async def process_review_workflow_comment():
    """Some docstring"""
    result = {some_var async for some_var in get_somethings()}
    return result
```


### Current behavior

Pylint reports that `get_somethings()` is used in an iterating context and is not iterable.

```
 pylint test.py
************* Module lel
test.py:10:45: E1133: Non-iterable value get_somethings() is used in an iterating context (not-an-iterable)

------------------------------------------------------------------
Your code has been rated at 1.67/10 (previous run: 1.67/10, +0.00)
```


### Expected behavior

Pylint does not report that `get_somethings()` is used in an iterating context and is not iterable.

### pylint --version output

```
 pylint --version
pylint 2.2.2
astroid 2.1.0
Python 3.7.1 (default, Nov  6 2018, 18:45:35)
[Clang 10.0.0 (clang-1000.11.45.5)]
```

--- a/CONTRIBUTORS.txt
+++ b/CONTRIBUTORS.txt
@@ -252,3 +252,6 @@ contributors:
 * Sergei Lebedev: contributor
 
 * Sasha Bagan
+
+* Pablo Galindo Salgado: contributor
+  Fix false positive 'Non-iterable value' with async comprehensions.

--- a/pylint/checkers/typecheck.py
+++ b/pylint/checkers/typecheck.py
@@ -1614,17 +1614,17 @@ class IterableChecker(BaseChecker):
     @check_messages("not-an-iterable")
     def visit_dictcomp(self, node):
         for gen in node.generators:
-            self._check_iterable(gen.iter)
+            self._check_iterable(gen.iter, check_async=gen.is_async)
 
     @check_messages("not-an-iterable")
     def visit_setcomp(self, node):
         for gen in node.generators:
-            self._check_iterable(gen.iter)
+            self._check_iterable(gen.iter, check_async=gen.is_async)
 
     @check_messages("not-an-iterable")
     def visit_generatorexp(self, node):
         for gen in node.generators:
-            self._check_iterable(gen.iter)
+            self._check_iterable(gen.iter, check_async=gen.is_async)
 
 
 def register(linter):

--- a/pylint/test/functional/iterable_context_py36.py
+++ b/pylint/test/functional/iterable_context_py36.py
@@ -26,3 +26,16 @@ async def count_to(number):
     """ counts to n in async manner"""
     async for i in some_iter_func(number):
         print(i)
+
+
+async def gen_values(num_values):
+    for value in range(num_values):
+        yield value
+
+
+async def do_some_comprehensions():
+    sets = {elem async for elem in gen_values(10)}
+    lists = [elem async for elem in gen_values(10)]
+    dicts = {elem: elem async for elem in gen_values(10)}
+    gen = (elem async for elem in gen_values(10))
+    return sets, lists, dicts, gen

